TTFAutoHint upgrade fails

- [X] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [X] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [X] ran `brew update` and can still reproduce the problem?
- [X] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [X] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
https://gist.github.com/6efa53cd73bd39373052d91846a9ba3a
- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)
Trying to upgrade TTFAutoHint from 1.6 to 1.8.2, because it is the only remaining out of date package when running brew upgrade
- What happened (include command output)

    ==> Upgrading 1 outdated package:
    ttfautohint 1.6 -> 1.8.2
    ==> Upgrading ttfautohint --with-qt
    ==> Downloading https://downloads.sourceforge.net/project/freetype/ttfautohint/1.8.2/ttfautohint-1.8.2.tar.gz
    Already downloaded: /Users/xcaiver/Library/Caches/Homebrew/downloads/5d6093004d37d21251be7fac840fba291eb992d8546c9b795f4c51ebd723c3c3--ttfautohint-1.8.2.tar.gz
    ==> ./configure --disable-silent-rules --prefix=/usr/local/Cellar/ttfautohint/1.8.2 --without-doc
    Last 15 lines from /Users/xcaiver/Library/Logs/Homebrew/ttfautohint/01.configure:
    checking whether wcsstr is declared without a macro... yes
    checking whether wcstok is declared without a macro... yes
    checking whether wcswidth is declared without a macro... yes
    checking whether wcsftime is declared without a macro... yes
    checking for stdint.h... (cached) yes
    checking for HARFBUZZ... yes
    checking for qmake... /usr/local/opt/qt/bin/qmake
    checking for moc... /usr/local/opt/qt/bin/moc
    checking for uic... /usr/local/opt/qt/bin/uic
    checking for rcc... /usr/local/opt/qt/bin/rcc
    checking whether host operating system is Darwin... yes
    checking whether QMAKESPEC environment variable is set... no, using -spec macx-g++
    Info: creating stash file /private/tmp/ttfautohint-20181118-68764-4temow/ttfautohint-1.8.2/conftest.dir/.qmake.stash
    Project ERROR: failed to parse default search paths from compiler output
    configure: error: Calling /usr/local/opt/qt/bin/qmake -spec macx-g++ failed.

- What you expected to happen
For the upgrade to succeed...
- Step-by-step reproduction instructions (by running `brew install` commands)
brew upgrade ttfautohint

https://gist.github.com/6efa53cd73bd39373052d91846a9ba3a

The --with-qt is coming from 'somewhere'... The actual command I entered was just '_brew upgrade ttfautohint_'

Even if I enter '_brew upgrade ttfautohint --without-qt_' (which should be a valid parameter) that gets stripped and it still adds the --with-qt back in:

 ```
 ~    brew upgrade ttfautohint --without-qt
==> Upgrading 1 outdated package:
ttfautohint 1.6 -> 1.8.2
==> Upgrading ttfautohint --with-qt
==> Downloading https://downloads.sourceforge.net/project/freetype/ttfautohint/1.8.2/ttfautohint-1.8.2.tar.gz
Already downloaded: /Users/xcaiver/Library/Caches/Homebrew/downloads/5d6093004d37d21251be7fac840fba291eb992d8546c9b795f4c51ebd723c3c3--ttfautohint-1.8.2.tar.gz
==> ./configure --disable-silent-rules --prefix=/usr/local/Cellar/ttfautohint/1.8.2 --without-doc
Last 15 lines from /Users/xcaiver/Library/Logs/Homebrew/ttfautohint/01.configure:
checking whether wcsstr is declared without a macro... yes
checking whether wcstok is declared without a macro... yes
checking whether wcswidth is declared without a macro... yes
checking whether wcsftime is declared without a macro... yes
checking for stdint.h... (cached) yes
checking for HARFBUZZ... yes
checking for qmake... /usr/local/opt/qt/bin/qmake
checking for moc... /usr/local/opt/qt/bin/moc
checking for uic... /usr/local/opt/qt/bin/uic
checking for rcc... /usr/local/opt/qt/bin/rcc
checking whether host operating system is Darwin... yes
checking whether QMAKESPEC environment variable is set... no, using -spec macx-g++
Info: creating stash file /private/tmp/ttfautohint-20181119-70557-13tkl74/ttfautohint-1.8.2/conftest.dir/.qmake.stash
Project ERROR: failed to parse default search paths from compiler output
configure: error: Calling /usr/local/opt/qt/bin/qmake -spec macx-g++ failed.

READ THIS: https://docs.brew.sh/Troubleshooting

These open issues may also help:
TTFAutoHint upgrade fails https://github.com/Homebrew/homebrew-core/issues/34257

 ~  
```

I worry that if I uninstall the older version and clean install the newer version that I'll lose the repro for the bug, even if I uninstall/reinstall the older version directly again

--- a/Formula/ttfautohint.rb
+++ b/Formula/ttfautohint.rb
@@ -21,36 +21,22 @@ class Ttfautohint < Formula
     depends_on "pkg-config" => :build
   end
 
-  option "with-qt", "Build ttfautohintGUI also"
-
-  deprecated_option "with-qt5" => "with-qt"
-
   depends_on "pkg-config" => :build
   depends_on "freetype"
   depends_on "harfbuzz"
   depends_on "libpng"
-  depends_on "qt" => :optional
 
   def install
-    args = %W[
-      --disable-dependency-tracking
-      --disable-silent-rules
-      --prefix=#{prefix}
-      --without-doc
-    ]
-
-    args << "--without-qt" if build.without? "qt"
-
     system "./bootstrap" if build.head?
-    system "./configure", *args
+    system "./configure", "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}",
+                          "--without-doc",
+                          "--without-qt"
     system "make", "install"
   end
 
   test do
-    if build.with? "qt"
-      system "#{bin}/ttfautohintGUI", "-V"
-    else
-      system "#{bin}/ttfautohint", "-V"
-    end
+    system "#{bin}/ttfautohint", "-V"
   end
 end

